src/context/AppProvider/hooks/useAccountsFetch.tsx

```diff
@@ -26,6 +26,9 @@ export const useAccountsFetch = () => {
   // Fetch portfolio for all managed accounts as a side-effect if they exist instead of going through the initial account detection flow.
   // This ensures that we have fresh portfolio data, but accounts added through account management are not accidentally blown away.
   useEffect(() => {
+    // Do not fetch accounts if the wallet modal is open or we're reconciliating local wallet - user is either inputting password, switching accounts, or their wallet is being rehydrated
+    if (modal || isLoadingLocalWallet) return
```

Comment:
Probably the worst of the bunch... We were refetching accounts while in a modal opened / rehydration state which was a race condition waiting to happen.

Reply:
So the thing is the wallet *is* connected but rehydrating 

Basically would go something like `isConnected -> !isConnected` as we're rehydrating with `isLoadingLocalWallet` (forgot the exact flow) 

`modal` is more of a paranoia tbh, `isLoadingLocalWallet` is really the one we're after